### PR TITLE
Fix timeout and size restrictions handling for snapshot in odsp driver

### PR DESCRIPTION
**Pull request by jatgarg**
1.) Respect the host policy if specified.
2.) Remove size restriction for summarizer client.
3.) If the snpashot timeout or size was too big and we initially fetched with blobs, then try fetching without blobs.
4.) Clear the timeout if it goes out.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 1]
```diff
@@ -539,8 +541,9 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
             const odspSnapshot = await this.fetchSnapshotCore(snapshotOptions, tokenFetchOptions, usePost, abortController);
             return odspSnapshot;
         } catch (error) {
-            if (error.errorType === OdspErrorType.snapshotTooBig || error.errorType === OdspErrorType.fetchTimeout) {
-                const snapshotOptionsWithoutBlobs: ISnapshotOptions = { ...snapshotOptions, blobs: undefined };
+            // If the first snapshot request was with blobs and we either timed out or the size was too big, then try to fetch without blobs.
+            if ((error.errorType === OdspErrorType.snapshotTooBig || error.errorType === OdspErrorType.fetchTimeout) && snapshotOptions.blobs) {
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:545]: if it's OdspErrorType.snapshotTooBig, but user provided their own mds setting (that is smaller than our default one), then we should throw that error to user.

Comment (vladsud) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:545]: This need to be fixed - that would break Teams - they limit it at 1Mb or so and do not want to load file if it's bigger.

Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:553]: Added the check for the same.

**Review by vladsud — APPROVED** [review]
Looks better!
I think I'd go even further. That's minor, so not blocking merge.
If host says - limit snapshot at 50Mb, but we limit at 25Mb and hit that limit, I do not thing we want to go with just trees, but otherwise unlimited bandwidth (we would download whole file at worst). 
If host says - 50Mb is the limit, then I think we should just use it (and not overwrite anything), and then propagate error.

In other words:
1. Move mds up such that user policy overwrites our defaults
2. If we fail on size, and size was provided by user, throw an error.
